core/src/autogluon/core/space.py

```diff
@@ -123,6 +127,14 @@ def convert_to_sklearn(self):
             sampler = uniform(self.lower, self.upper - self.lower)
         return sampler
 
+    def convert_to_ray(self):
```

Comment:
Design concerns: `Space` should not be aware of all possible ways to convert it. Conversion strategy should be decoupled from the entity. See [adapter pattern](https://refactoring.guru/design-patterns/adapter/python/example).
I would expect that we will pass `Space` to Ray-enabled sub-code, where it would be wrapped into Ray-specific adapter, which knows how to translate `Space` into ray spaces.

Reply:
Good point!

Reply:
I guess this is not a strict adapter pattern because we don't want to implement the adapter for ray tune search space interface. This is more like a converter, but I can make it to be a standalone object instead of embedded in `Space` so that it's decoupled